sds_data_manager/lambda_code/efs_lambda/lambda_function.py

```diff
@@ -0,0 +1,140 @@
+import os
+import shutil
+
+import boto3
+
+# Define the paths
+mount_path = os.environ.get("EFS_MOUNT_PATH")
+
+attitude_symlink_path = f"{mount_path}/latest_attitude_kernel.ah.a"
+ephemeris_symlink_path = f"{mount_path}/latest_ephemeris_kernel.bsp"
+
+
+def create_symlink(source_path: str, destination_path: str) -> None:
+    """Create a symlink from source_path to destination_path
+
+    Parameters
+    ----------
+    source_path : str
+        Source path of the symlink
+    destination_path : str
+        Destination path of the symlink
+    """
+
+    # Check if the symlink exists
+    if os.path.islink(destination_path):
+        # Remove the old symlink
+        os.unlink(destination_path)
+
+    # Create a new symlink pointing to the new file
+    os.symlink(source_path, destination_path)
+
+
+def write_data_to_efs(s3_key: str, s3_bucket: str):
+    """Write data to EFS and create/update symlink
+
+    Parameters
+    ----------
+    s3_key : str
+        S3 object key
+    s3_bucket : str
+    """
+    filename = os.path.basename(s3_key)
+    # Set the download path in the /tmp directory
+    download_path = "/tmp/" + os.path.basename(s3_key)
+    # Create an S3 client
+    s3_client = boto3.client("s3")
+
+    try:
+        # Download the file from S3
+        s3_client.download_file(s3_bucket, s3_key, download_path)
+        print(f"File downloaded: {download_path}")
+    except Exception as e:
+        print(f"Error downloading file: {e!s}")
+
+    # Move the file to the mount directory. Eg. /mnt/efs
+    destination_path = f"{mount_path}/" + os.path.basename(s3_key)
+    try:
+        shutil.move(download_path, destination_path)
+        print(f"File moved to: {destination_path}")
+    except Exception as e:
+        print(f"Error moving file: {e!s}")
+
+    print("After : ", os.listdir(mount_path))
+
+    # Attitude naming convention is this:
+    # imap_yyyy_doy_yyyy_doy_##.ah.bc and
+    # imap_yyyy_doy_yyyy_doy_##.ah.a
+    if filename.endswith(".ah.a") or filename.endswith(".ah.bc"):
+        create_symlink(filename, attitude_symlink_path)
+
+    # Ephemeris naming convention is this:
+    # Eg. imap_nom_yyyymmdd_yyyymmdd_v##.bsp
+    # The reason we check startswith is because other ephemeris
+    # kernel has different prefix. Eg.
+    #   imap_recon_yyyymmdd_yyyymmdd_v##.bsp
+    #   imap_burn_yyyymmdd_yyyymmdd_v##.bsp
+    #   imap_pred_yyyymmdd_yyyymmdd_v##.bsp
+    elif filename.startswith("imap_nom") and filename.endswith(".bsp"):
+        create_symlink(filename, ephemeris_symlink_path)
+
+    print("Done with symlink creation")
+
+    print("After : ", os.listdir(mount_path))
```

Comment:
Remove these debugs? Or change to logging if you think we'll need it.

Reply:
I will remove them. It was helpful for development.

Reply:
removed and changed some to logging debug level